irepela/homework_10/iterators.py

```diff
@@ -0,0 +1,70 @@
+from collections.abc import Iterator
+
+
+class SimpleIterator(Iterator):
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __next__(self):
+        if self.index >= len(self.iterable):
+            raise StopIteration
+        next_item = self.iterable[self.index]
+        self.index += 1
+        return next_item
+
+
+class ReversedIterator(Iterator):
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = len(iterable) - 1
+
+    def __next__(self):
+        if self.index < 0:
+            raise StopIteration
+        next_item = self.iterable[self.index]
+        self.index -= 1
+        return next_item
+
+
+class CycleIterator(Iterator):
+
+    def __init__(self, iterable):
+        self.iterable = iterable
+        self.index = 0
+
+    def __next__(self):
+        while True:
```

Comment:
you do not need while here

you return in try
you break in except

you will never have 2 turn

Reply:
Removed while loop